dotCMS/src/main/java/com/dotmarketing/business/PermissionBitAPIImpl.java

```diff
@@ -579,34 +578,39 @@ public void copyPermissions(Permissionable from, Permissionable to) throws DotDa
 	 * @param id of permission to find
 	 * @return  Permission
 	 */
+	@CloseDBIfOpened
 	public  Permission find(String id){
 		return permissionFactory.getPermission(String.valueOf(id));
 	}
 
-
+	@CloseDBIfOpened
 	public List<Permission> getPermissions(Permissionable permissionable) throws DotDataException {
 		return permissionFactory.getPermissions(permissionable, false);
 	}
 
+	@CloseDBIfOpened
 	public List<Permission> getPermissions(Permissionable permissionable, boolean bitPermissions) throws DotDataException {
 		return permissionFactory.getPermissions(permissionable, bitPermissions);
 	}
 
+	@CloseDBIfOpened
 	public List<Permission> getPermissions(Permissionable permissionable,
 			boolean bitPermissions, boolean onlyIndividualPermissions) throws DotDataException {
 		return permissionFactory.getPermissions(permissionable, bitPermissions, onlyIndividualPermissions);
 	}
 
-
+	@CloseDBIfOpened
 	public List<Permission> getPermissions(Permissionable permissionable,
 			boolean bitPermissions, boolean onlyIndividualPermissions, boolean forceLoadFromDB) throws DotDataException {
 		return permissionFactory.getPermissions(permissionable, bitPermissions, onlyIndividualPermissions, forceLoadFromDB);
 	}
 
+	@CloseDBIfOpened
     public void addPermissionsToCache ( Permissionable permissionable ) throws DotDataException {
         permissionFactory.addPermissionsToCache( permissionable );
     }
-    
+
+    // todo: should be this a transaction (all of nothing on save several permissions)???

```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)